### ISSUE TITLE
'Audio Device Could not be opened' error with Jack

### ISSUE THREAD

**Issue by Holzhaus** [Holzhaus/mixxx-gh-issue-migration#128]
Reported by: jukingeo
Date: 2008-07-13T19:14:34Z
Status: Won't Fix
Importance: Low
Launchpad Issue: [lp248202](https://bugs.launchpad.net/bugs/248202)
Tags: jack

    When Jack Audio Connection Kit is selected for the sound device under Preferences, a pop up box comes up with:

    Configuration Error: Audio Device Could Not Be Opened.

    However, when MIXXX is set to ALSA for the audio device, it does work, albeit with a very crackly and distorted sound.  (The distortion is a separate issue).

    The error occurs EVERY time Jack is started and then audio preferences is set to JACK.

    It is understood that Jack must be started prior to the start up of MIXXX.

**Comment by Holzhaus** [comment]
Created by: gamegod
Date: 2008-07-13T23:02:42Z

    So you've run jackd before starting Mixxx? ... and you've changed the Sound API in the preferences to JACK?

    ... and the samplerate you have selected matches the one JACK picked? (Try 48000...)

**Comment by Holzhaus** [comment]
Created by: jukingeo
Date: 2008-07-15T01:36:45Z

    On Sun, 2008-07-13 at 23:02 +0000, Albert Santoni wrote:

    > <<So you've run jackd before starting Mixxx?>>

    Yup!

    > <<... and you've changed the
    > Sound API in the preferences to JACK?>>

    Yup
    > 
    > <<... and the samplerate you have selected matches the one JACK picked?
    > (Try 48000...)>>

    That worked!  BUT we still have a problem.  If I play two files at one
    time and make adjustments, I get xruns out the yin yang.

    Also I think it would be a good idea that if the sample rates don't
    match to change the error message to:

    ""Audio Device Could Not be Opened:
    Check to see if sample rates match.""

    I think adding that extra line will save you guys a lot of grief when
    people can't get their Jack Control to work.

    However, as I pointed out above, there is still a problem with Jack as
    the program causes xruns.  I don't have the problem with other
    software...except for another DJ program called DJPlay.  That too gave
    me xrun problems.  But TerminatorX another DJ program works fine.

    Thanx,

    Geo

**Comment by Holzhaus** [comment]
Created by: enrydj
Date: 2008-08-17T12:39:43Z

    Hi, I have a similar problem: with normal stereo settings jack seems to work well, but sometimes tracks don't play when pressing play button after they are loaded into decks. If from preferencies I select "ALSA" and then "Jack" again, sometimes track that didn't play begins to play. When I select "jack" again, sometimes 

    Configuration Error: Audio Device Could Not Be Opened.

    appears to me, too. See my log, I hope this can be useful.

    I launch Jack like this: jackd -R -P 10 --d alsa -o 2 -P -d hw:0 -r 44100 -S -p512

**Comment by Holzhaus** [comment]
Created by: enrydj
Date: 2008-08-17T14:15:44Z

**Comment by Holzhaus** [comment]
Created by: raskolnikov
Date: 2010-09-14T15:12:15Z

    I had the same issue and found that when loading Mixxx takes long to reply to jackd and then it timeouts and gets disconnected. The soultion, in "qjackctl" settings: Increase the parameter "Timeout (msec)" to high number (I did 6000).

**Comment by Holzhaus** [comment]
Created by: @ywwwg
Date: 2011-10-05T14:10:25Z

    These sound like classic JACK problems.  I agree with raskolnikov, make sure your timeout is set higher.  2000 works for me.  also try --nozombies.  I would say this isn't our bug unless you have flawless JACK performance in other apps (ardour2, jack-rack, xwax) and bad performance in mixxx.

**Comment by Holzhaus** [comment]
Created by: @bkgood
Date: 2011-10-08T18:50:55Z

    FYI the sound preferences should now be giving error messages as suggested. Is there still a bug here?

**Comment by Holzhaus** [comment]
Issue closed with status "Won't Fix".
